sw/host/tests/chip/jtag/src/test_sram_load.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fs;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::time::Duration;
+
+use anyhow::Result;
+use regex::Regex;
+use structopt::StructOpt;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::io::jtag::{JtagParams, JtagTap};
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::load_sram_program;
+use opentitanlib::uart::console::UartConsole;
+use opentitanlib::util::vmem::Vmem;
+
+// use top_earlgrey::top_earlgrey_memory;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+    #[structopt(flatten)]
+    jtag: JtagParams,
+    /// Path to the program to load
+    #[structopt(long)]
+    vmem: PathBuf,
+}
+
+fn test_sram_load(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    //
+    // Connect to the RISC-V TAP
+    //
+    transport.pin_strapping("PINMUX_TAP_RISCV")?.apply()?;
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+
+    let jtag = transport.jtag(&opts.jtag)?;
+    log::info!("Connecting to RISC-V TAP");
+    jtag.connect(JtagTap::RiscvTap)?;
+    log::info!("Halting core");
+    jtag.halt()?;
+    log::info!("Loading program to execute");
+    let vmem_content = fs::read_to_string(&opts.vmem)?;
+    let mut vmem = Vmem::from_str(&vmem_content)?;
+    vmem.merge_sections();
```

Comment:
it would be nice if we moved this to the `load_sram_program` function so it happens under the hood, wdyt?